src/utils/metadata.ts

```diff
@@ -52,9 +52,12 @@ export async function getPublisherFromServiceSD(
   serviceSD: any
```

Comment:
suggestion(blocking): Would like to have the scope of this function be more defined / typed:
```ts
export type ServiceSelfDescription = {
  selfDescriptionCredential: {
    credentialSubject: {
      'gx-service-offering:providedBy': {
        '@value': string
      }
    }
  }
}

export async function getPublisherFromServiceSD(
  serviceSD: ServiceSelfDescription
) { 
// ...
}
```

And then calling the function accordingly (if raw data is given):
```ts
const parsed: ServiceSelfDescription = JSON.parse(raw)
getPublisherFromServiceSD(parsed)
```

Reply:
But what about all the other keys that are present in the service SD object? Do we just ignore them? I would actually like to update all the SD-related types, It would be pretty cool to have a package with all the compliance API types to just import the updated ones (like we are doing with `@oceanprotocol/lib`)

Reply:
For now the can just be ignored, we only need to type the ones that we actually use. A lib to import into projects could indeed be useful. Might be somthing to consider for the GX Lab.